lib/pbench/server/api/resources/query_apis/metadata_index/timeseries_samples.py

```diff
@@ -0,0 +1,232 @@
+
+        es_filter = [{"match": {"run.id": run_id}}]
+        query_strings = []
+
+        # Validate each user provided filters against the result-data document
+        # mappings. If the filter is not present in result-data mappings
+        # properties we will not apply that filter in the ES query.
+        for filter, value in json_data.get("filters", {}).items():
+            if filter in mappings.keys() and mappings[filter] == "keyword":
+                # Get all the keyword filters to apply
+                es_filter.append({"match": {filter: value}})
+            elif filter in mappings.keys() and mappings[filter] != "keyword":
+                # Get all the text filters to apply
+                # Note: There is only one text field sample.measurement_title
+                # in result-data documents and if we can re-index it as a
+                # keyword we can get rid of this loop.
+                query_strings.append(
+                    {"query_string": {"fields": f"{filter}", "query": f"{value}"}}
+                )
+
+        return {
+            "path": f"/{indices}/_search?scroll={TimeSeriesSamples.SCROLL_EXPIRY}",
+            "kwargs": {
+                "json": {
+                    "size": TimeSeriesSamples.DOCUMENT_SIZE,
+                    "query": {"bool": {"filter": es_filter, "must": query_strings}},
+                    "sort": [
+                        {"sample.start": {"order": "asc", "unmapped_type": "long"}}
+                    ],
+                },
+                "params": {"ignore_unavailable": "true"},
+            },
+        }
+
+    def postprocess(self, es_json: JSON, context: CONTEXT) -> Response:
+        """
+        Returns a Flask Response containing a JSON object with keys as
+        results and possibly a scroll_id if the next page of results
+        is available.
+
+        If there are more than 10,000 documents available for the given filters, then
+        we return the first 10,000 documents along with a scroll id which the client
+        can use to request the next 10,000 documents.
+
+        If there are no more than 10000 documents then we only return results
+        without any scroll id.
+
+        Example:
+            {
+                "scroll_id": "Scroll_id_string", # conditional
+                "results": [
+                    {
+                        "@timestamp": "2020-09-03T01:58:58.712889",
+                        "@timestamp_original": "1626277442035",
+                        "result_data_sample_parent": "d8ec0b9d6affc5b638169db8b4cbbda2",
+                        "run": {...},
+                        "iteration": {...},
+                        "sample": {...},
+                        "result": {...}
+                    },
+                    {
+                        "@timestamp": "2021-03-03T01:58:58.712889",
+                        "@timestamp_original": "1626277442038",
```

Comment:
I realize that this is just sample data, but these values are rather amusing:  we have two timeseries values (in the `@timestamp` fields) which differ, down to the _microsecond_, by exactly six **months**...that was a very long and very precise run!  😁  Also, the values in the `@timestamp_original` fields, which look more reasonable (i.e., they differ from each other by three milliseconds), bear no resemblance to the values in the `@timestamp` fields.  🙂